crates/octez/src/async/client.rs

```diff
@@ -331,6 +332,83 @@ impl OctezClient {
         self.spawn_and_wait_command(args).await?;
         Ok(())
     }
+
+    pub async fn originate_contract(
+        &self,
+        contract_alias: &str,
+        funding_account_src: &str, // can be an address or an alias
+        fund_tez: f64,
+        contract_path: &Path,
+        init_data: Option<&str>,
+        burn_cap: Option<f64>,
+    ) -> Result<(ContractKt1Hash, OperationHash)> {
+        let fund_str = fund_tez.to_string();
+        let burn_cap_str = match burn_cap {
+            Some(v) => v.to_string(),
+            None => String::new(),
+        };
+        let mut args = vec![
+            "originate",
+            "contract",
+            contract_alias,
+            "transferring",
+            &fund_str,
+            "from",
+            funding_account_src,
+            "running",
+            contract_path
+                .to_str()
+                .ok_or(anyhow::anyhow!("failed to convert contract path to string"))?,
+        ];
+        if let Some(data) = init_data {
+            args.append(&mut vec!["--init", data]);
+        }
+        if !burn_cap_str.is_empty() {
+            args.append(&mut vec!["--burn-cap", &burn_cap_str]);
+        }
+        let output = self.spawn_and_wait_command(args).await?;
+
+        let operation_hash = parse_operation_hash(&output);
+        if operation_hash.is_none() {
+            anyhow::bail!("failed to parse operation hash from execution output");
+        }
+        let contract_address = parse_contract_address(&output);
+        if contract_address.is_none() {
+            anyhow::bail!("failed to parse contract address from execution output");
+        }
+        Ok((contract_address.unwrap(), operation_hash.unwrap()))
+    }
+}
+
+fn parse_regex(pattern_str: &str, output: &str) -> Option<String> {
+    let pattern = regex::Regex::new(pattern_str).unwrap();
```

Comment:
```suggestion
    let pattern = regex::Regex::new(pattern_str).ok()?;
```

Reply:
Updated

Reply:
Hmm in that case, would it be useful to return Result instead of Option because the caller of `parse_` expects the output to contain the regex and it should be a failure otherwise.

Reply:
Okay I've changed this back to Result. I thought not finding anything should not be an error from the point of view of `parse_regex` itself, but I suppose in this case specifically we can consider that an error.